src/content_delegate.js

```diff
@@ -585,31 +588,193 @@ ContentDelegate.prototype.attachRecapLinkToEligibleDocs = function() {
+    const page = document.createElement("html");
+    page.innerHTML = html;
+    const frames = page.querySelectorAll("iframe");
+    return frames[0].src;
+  };
+
+  // helper function - convert string to html document
+  const stringToDocBody = (str) => {
+    const parser = new DOMParser();
+    const newDoc = parser.parseFromString(str, 'text/html');
+    return newDoc.body;
+  };
+
+  // helper function - returns filename based on user preferences
+  const generateFileName = (options, pacerCaseId) => {
+    if (options.ia_style_filenames) {
+      return [
+        'gov',
+        'uscourts',
+        this.court,
+        (pacerCaseId || 'unknown-case-id')
+      ].join('.').concat('.zip');
+    } else if (options.lawyer_style_filenames) {
+      const firstTable = document.getElementsByTagName('table')[0];
+      const firstTableRows = firstTable.querySelectorAll('tr');
+      // 4th from bottom
+      const matchedRow = firstTableRows[firstTableRows.length - 4];
+      const cells = matchedRow.querySelectorAll('td');
+      const document_number = cells[0].innerText.match(/\d+(?=\-)/)[0];
+      const docket_number = cells[1].innerText;
+      return [
+        PACER.COURT_ABBREVS[this.court],
+        docket_number,
+        document_number,
+      ].join('_').concat('.zip');
+    }
+  };
+
+  // Make the Back button redisplay the previous page.
+  window.onpopstate = function (event) {
+    if (event.state.content) {
+      document.documentElement.innerHTML = event.state.content;
+    }
+  };
+  history.replaceState({content: document.documentElement.innerHTML}, '');
+
+  try {
+    // runtime start
+    $("body").css("cursor", "wait");
+    // fetch the html page which contains the <iframe> link to the zip document.
+
+    // in Firefox, use content.fetch for content-specific fetch requests
+    // https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts#XHR_and_Fetch
+    const browserSpecificFetch = (navigator.userAgent.indexOf('Chrome') < 0) ? content.fetch : window.fetch
+
+    const htmlPage = await browserSpecificFetch(event.data.id).then(res => res.text());
+    console.log("RECAP: Successfully submitted zip file request");
+    const zipUrl = extractUrl(htmlPage);
+
+    //download zip file and save it to chrome storage
+    const nonce = "blob_upload_storage";
+    const zipFile = await fetch(zipUrl)
+      .then(res => res.arrayBuffer())
+      .then(buffer => {
+        console.info('RECAP: Downloaded zip file');
+        saveItemToStorage({[nonce]: arrayBufferToArray(buffer)});
+        return new Blob([buffer], {type: 'application/zip'});
+      })
+    const blobUrl = URL.createObjectURL(zipFile);
+    const pacerCaseId = (event.data.id).match(/caseid\=\d*/)[0].replace(/caseid\=/, "");
+
+    // load options
+    const payload = await getItemsFromStorage(['options']);
+    const options = payload['options'];
+
+    if (options['recap_enabled'] && !this.restricted) {
```

Comment:
Can we flip this logic to make it an early abort from the function and de-indent some of this?

This is probably bad JS, but:
```suggestion
if (!options['recap_enabled'] || this.restricted) {
  return;
}

```

Reply:
Stylistic preference. Up to you.